### ISSUE TITLE
Improves component Data Provider polling engine

### ISSUE THREAD

**Issue by matuzalemsteles** [liferay/clay#2719]
@julien brought some interesting insights we can explore to make our Data Provider polling engine smarter. See https://github.com/liferay/clay/pull/2680#issuecomment-549809648

We can remove the use of `setTimeout` to use `requestAnimationFrame`, so we would have to create some scheduler implementation to work as expected.

**Comment by kresimir-coko** [comment]
Is this still a valid enhancement @matuzalemsteles ?

**Comment by kresimir-coko** [comment]
Is this still a valid enhancement @matuzalemsteles ?

**Comment by matuzalemsteles** [comment]
@kresimir-coko I would say yes, it would be a good improvement.

**Comment by bryceosterhaus** [comment]
@matuzalemsteles whats the status on moving the DataProvider to live in DXP? I remember a few weeks back we decided it was best to move DataProvider from clay to dxp

**Comment by matuzalemsteles** [comment]
@bryceosterhaus yeah, I mentioned that in another issue now, it was "deprioritized" but I think we can start working on it now, create an epic on the Jira and move those issues there.

**Comment by matuzalemsteles** [comment]
LPS Ticket https://liferay.atlassian.net/browse/LPS-190549
